src/Components/Continents.jsx

```diff
@@ -0,0 +1,160 @@
+/* eslint-disable no-plusplus */
+/* eslint-disable no-unused-vars */
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable no-unused-expressions */
+/* eslint-disable no-undef */
+/* eslint-disable no-tabs */
+/* eslint-disable no-mixed-spaces-and-tabs */
+/* eslint-disable max-len */
+/* eslint-disable react/destructuring-assignment */
+/* eslint-disable react/prop-types */
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  Link,
+  Routes,
+  Route,
+  useNavigate,
+  useLocation,
+} from 'react-router-dom';
+import { MdAdsClick } from 'react-icons/md';
+import { IoArrowBack } from 'react-icons/io5';
+import Countries from './Countries';
+import { fetchCountries } from '../myredux/myreducer';
+
+const Continents = (props) => {
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchCountries());
+  }, []);
+
+  // ! ----------------------------------------------------
+  const { name: continentIamAt } = props;
+
+  const mystore = useSelector((state) => state);
+
+  const [countriesIamAt, setCountriesIamAt] = useState([]);
+
+  useEffect(() => {
+    const tempCountries = mystore.continents.filter(
```

Comment:
- [ ] You should structure your project into containers then breaks containers into components. Kindly create a containers folder and ensure you break your containers into components.

Reply:
@mikenath223  
I am sorry I do not quite understand what you mean by containers because I did break down my app into components, and I do not have many reuseable components to break them any further.

Reply:
Ok then break it further into more reusable components. You have really large components that defeat the idea of components which should be small and reusable.

Reply:
sorry because of my state I can not do it. it will be more repetition than what it is right now.